### PR TITLE
ELEMENTS-730: add depth's notion to selectivity entries (Backport 9.10)

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

**Comment by guirenard** [comment]
:+1:

File: widgets/nuxeo-selectivity.html [hunk 1]
```diff
@@ -329,11 +329,17 @@
           templates: {
             resultItem: function(options) {
               return (
-                  '<div class="selectivity-result-item' +
-                  (options.disabled ? ' disabled' : '') +
-                  '" data-item-id="' + this.escapeHTML(options.id) + '">' +
-                  this.resultFormatter(options.item)
-                  + '</div>');
+                  '<div class="selectivity-result-item' + (options.disabled ? ' disabled' : '') +
+                  '" style="padding-left:' + (7 + (10 * options.depth)) + 'px" data-item-id="' +
```
Comment (Gabez0r) [inline widgets/nuxeo-selectivity.html]: nit: missing space after `padding-left:`:
```suggestion
                  '" style="padding-left: ' + (7 + (10 * options.depth)) + 'px" data-item-id="' +
```

File: widgets/nuxeo-selectivity.html [hunk 2]
```diff
@@ -329,11 +329,17 @@
...
+                  this.escapeHTML(options.id) + '">' + this.resultFormatter(options.item) + '</div>'
+              );
+            }.bind(this),
+
+            resultLabel: function(options) {
+              return (
+                    '<div class="selectivity-result-label" style="padding-left:' + (7 + (10 * options.depth)) + 'px">' +
```
Comment (Gabez0r) [inline widgets/nuxeo-selectivity.html]:
```suggestion
                    '<div class="selectivity-result-label" style="padding-left: ' + (7 + (10 * options.depth)) + 'px">' +
```

File: widgets/nuxeo-selectivity.html [hunk 3]
```diff
@@ -513,19 +519,21 @@
...
           };
           if (item.children) {
-            entry.children = item.children.map(fn.bind(this));
+            entry.children = item.children.map(function(child) { return fn(child, entry.depth + 1) });
           }
           return entry;
-        };
-        return Array.isArray(value) ? value.map(fn.bind(this)) : fn.call(this, value);
+        }.bind(this);
+        return Array.isArray(value) ? value.map(function(item) { return fn(item) }) : fn(value);
```
Comment (Gabez0r) [inline widgets/nuxeo-selectivity.html]: Passing `fn` to the map should do the same:
```suggestion
        return Array.isArray(value) ? value.map(fn) : fn(value);
```

Comment (nmpcunha) [reply widgets/nuxeo-selectivity.html]: Added a comment explaining why it does not the same 👍

File: widgets/nuxeo-selectivity.html [hunk 4]
```diff
@@ -513,19 +519,25 @@
...
           var entry = {
             id: id,
             text: item.displayLabel || item.text || id || '',
-            item: item
+            item: item,
+            depth: depth
           };
           if (item.children) {
-            entry.children = item.children.map(fn.bind(this));
+            entry.children = item.children.map(function(child) { return fn(child, entry.depth + 1) });
```
Comment (Gabez0r) [inline widgets/nuxeo-selectivity.html]: nit: missing semicolon at the end of return statement:
```suggestion
            entry.children = item.children.map(function(child) { return fn(child, entry.depth + 1); });
```

File: widgets/nuxeo-selectivity.html [hunk 5]
```diff
@@ -513,19 +519,25 @@
...
           }
           return entry;
-        };
-        return Array.isArray(value) ? value.map(fn.bind(this)) : fn.call(this, value);
+        }.bind(this);
+        /*
+         * XXX: Since array methods like map and forEach takes index as second parameter
+         * we forced the function call instead of calling something like array.map(fn);
+         */
+        return Array.isArray(value) ? value.map(function(item) { return fn(item) }) : fn(value);
```
Comment (Gabez0r) [inline widgets/nuxeo-selectivity.html]:
```suggestion
        return Array.isArray(value) ? value.map(function(item) { return fn(item); }) : fn(value);
```

**Comment by Gabez0r** [comment]
👍

**Pull request merged**